cfme/utils/net.py

```diff
@@ -58,6 +58,51 @@ def ip_echo_socket(port=32123):
             conn.close()
 
 
+def pick_responding_ip(vm, port, num_sec, rounds_delay_second, attempt_timeout):
+    """
+    Given a vm and port, pick one of the vm's addresses that is connectible
+    on the given port
+
+    Args:
+        vm: mgmt vm
+        port: port number to attempt connecting to
+        num_sec: Minimal ammount of time how long to keep checking (slight
+                 variation may happen -- approximately the attempt_timeout).
+        rounds_delay_second: The delay to wait after checking each IP round in
+                             immediate succession.
+        attempt_timeout: A connection timeout for every connection attempt.
+
+    Raise TimedOutError if no such IP is found.
+    """
+
+    def connection_factory(ip):
+        if net_check(port, ip, attempt_timeout):
+            return ip
+
+    return retry_connect(vm, connection_factory, num_sec, rounds_delay_second)
+
+
+def retry_connect(ips_getter, connection_factory, num_sec, delay):
+    def try_batch_of_ips():
```

Comment:
since this method is internal: 
```suggestion
    def _try_batch_of_ips():
```

Reply:
I don't understand the reasoning for this much, but I really don't care about the underscore. I am changing this.